lib/server.js

```diff
@@ -28,6 +28,12 @@ function createServer (options, httpHandler) {
   } else {
     server = http.createServer(httpHandler)
     server.keepAliveTimeout = options.keepAliveTimeout
+    // we treat zero as null
+    // and null is the default setting from nodejs
+    // so we do not pass the option to server
+    if (options.maxRequestsPerSocket > 0) {
+      server.maxRequestsPerSocket = options.maxRequestsPerSocket
+    }
```

Comment:
For all of you reading this, I just opened https://github.com/nodejs/node/issues/40190 because of it. This if should not be needed.